pkg/virt-controller/watch/preset_test.go

```diff
@@ -884,19 +847,22 @@ var _ = Describe("VirtualMachineInstance Initializer", func() {
 			vmi := v1.NewMinimalVMI("testvmi")
 			vmi.Labels = map[string]string{flavorKey: presetFlavor}
 
-			// Register the expected REST call
-			server.AppendHandlers(
-				ghttp.CombineHandlers(
-					ghttp.VerifyRequest("PUT", "/apis/kubevirt.io/v1alpha2/namespaces/default/virtualmachineinstances/testvmi"),
-					ghttp.RespondWithJSONEncoded(http.StatusOK, vmi),
-				),
-			)
+			virtClient.EXPECT().VirtualMachineInstance(k8smetav1.NamespaceDefault).Return(vmiInterface)
 
-			err := app.vmiPresetController.initializeVirtualMachine(vmi)
+			vmiInterface.EXPECT().Update(gomock.Any()).Do(func(arg interface{}) {
+				vmi := arg.(*v1.VirtualMachineInstance)
+				val, found := vmi.Annotations["virtualmachinepreset.kubevirt.io/test-preset"]
+				Expect(found).To(BeTrue(), "preset should have been applied")
+				Expect(val).To(Equal("kubevirt.io/v1alpha2"))
```

Comment:
I see, you're calling the controller.initializeVirtualMachine() and passing the vmi object directly in instead of controller.Execute()

Is there a reason we can't follow the same pattern in vm_test.go where the VM key is added to the queue and controller.Execute() is the entry point for the test?

Reply:
Well that codepath is exercised directly a couple of times actually. But for the rest of the tests I took the more direct (and readable) approach of calling `initializeVirtualMachine` directly.

That said. I think you're right. there's no value in testing the values twice as they don't differ for any of the cases calling `initializeVirtualMachine`.